src/backupNg/backupNg.spec.js

```diff
@@ -169,5 +188,61 @@ describe("backupNg", () => {
       });
       expect(log.warnings).toMatchSnapshot();
     });
+
+    it("executes two times a rolling snapshot with 2 as retention & revert to an old state", async () => {
+      jest.setTimeout(1e4);
+      const scheduleTempId = randomId();
+      const { id: jobId } = await xo.createTempBackupNgJob({
+        ...defaultBackupNg,
+        schedules: {
+          [scheduleTempId]: {
+            name: "scheduleTest",
+            cron: "0 * * * * *",
+          },
+        },
+        settings: {
+          [scheduleTempId]: { snapshotRetention: 2 },
+        },
+      });
+
+      const backupNgJob = await xo.call("backupNg.getJob", { id: jobId });
+      const settingKeys = Object.keys(backupNgJob.settings);
```

Comment:
To not repeat this piece of code each time we need to create a job with an associated schedule, you can create a function which get the schedule id from the job settings.